neuro_san/service/main_loop/server_status.py

```diff
@@ -21,27 +24,9 @@ def __init__(self, server_name: str):
         Constructor.
         """
         self.server_name: str = server_name
-        self.grpc_service_ready: bool = False
-        self.http_service_ready: bool = False
-        self.updater_ready: bool = False
-
-    def set_grpc_status(self, status: bool):
-        """
-        Set the status of gRPC service
-        """
-        self.grpc_service_ready = status
-
-    def set_http_status(self, status: bool):
-        """
-        Set the status of http service
-        """
-        self.http_service_ready = status
-
-    def set_updater_status(self, status: bool):
-        """
-        Set the status of dynamic agents registry updater
-        """
-        self.updater_ready = status
+        self.grpc_service: ServiceStatus = ServiceStatus("gRPC")
+        self.http_service: ServiceStatus = ServiceStatus("http")
```

Comment:
Could this instead eventually be a dictionary keyed by name? (Fine as is for now)
I think of the native MCP server case, for instance.